frontend/src/pages/Search.js

```diff
@@ -24,75 +24,102 @@ export default function Search() {
-    headers:{
-      'Content-type':'application/json',
-      'Authorization':`Bearer ${cookie}`
-    }
-  })
-  const data=await response.json();
-  SetisLoading(false);
-  data.users.length=data.users.length>2?data.users.length=2:data.users.length;
-  SetUsers(data.users)
-  if(data.users.length===0)
-  setResultsEmpty(true)
-  else
-  setResultsEmpty(false);
-
-  }
-
-    const accessChatHandler=(values)=>{
+    const cookie = localStorage.getItem("jwt");
+    const response = await fetch(
+      `${process.env.REACT_APP_API_URL}/api/v1/users?search=${value}`,
+      {
+        headers: {
+          "Content-type": "application/json",
+          Authorization: `Bearer ${cookie}`,
+        },
+      }
+    );
+    const data = await response.json();
+    SetisLoading(false);
+    data.users.length =
+      data.users.length > 2 ? (data.users.length = 2) : data.users.length;
+    SetUsers(data.users);
+    if (data.users.length === 0) setResultsEmpty(true);
+    else setResultsEmpty(false);
+  };
 
-      const isPresent=state.find((data)=>{
-        return data.email===values.email
-      });
-      dispatch(AddUser(values,state))
+  const accessChatHandler = (values) => {
+    let isPresent = false;
+    let activeChat = null;
+    for (let i = 0; i < allChats.length; i++) {
+      if (allChats[i].chatName === "sender") {
+        for (let j = 0; j < allChats[i].users.length; j++) {
+          if (allChats[i].users[j].email === values.email) {
+            isPresent = true;
+            activeChat = allChats[i];
+            break;
+          }
+        }
+      }
+      if (isPresent) {
+        break;
+      }
+    }
+    if (isPresent) {
+      dispatch(SetActiveChat(activeChat));
+      navigate("/home/message",{replace:true})
+    }else{
+      dispatch(AddUser(values, allChats));
       notify(values.name);
       setTimeout(()=>{
+        dispatch(SetActiveChat(allChats[0]));
         navigate('/home/message',{replace:true})
       },2000)
     }
+  };
 
   return (
-    <div className='w-[80vw] relative flex flex-col'>
-      <ToastContainer/>
-        <SearchBar onChange={onChangeTextHandler} searchHandler={searchHandler} ></SearchBar>
-        <div className=' w-[100%] flex box-border justify-center py-2 relative'>
-        {!isLoading&&resultsEmpty&&<p>0 matching results found</p>}
-       {isLoading&&<Loading></Loading>}
-       {!isLoading&&users.length>0&&( <div className='w-[60%] border-[1px] rounded-md border-[#acacac] px-[1%] py-[1%] flex flex-col'>
-{users.map((data,index)=><User accessChat={accessChatHandler} values={data} key={index}></User>)}
-</div>)}
-    </div>
+    <div className="w-[80vw] relative flex flex-col">
+      <ToastContainer />
+      <SearchBar
+        onChange={onChangeTextHandler}
+        searchHandler={searchHandler}
+      ></SearchBar>
+      <div className=" w-[100%] flex box-border justify-center py-2 relative">
+        {!isLoading && resultsEmpty && <p>0 matching results found</p>}
```

Comment:
Try refactoring this part, instead of using isLoading so many times, try to figure out a way, it is used less times, so code looks more organized.

Reply:
Yes You can improve this

Reply:
tbh, @Spiral-Memory I didn't changed that part at all. The changes were showing due to formatting the document 😅.  But, I'll try to fix it !

Reply:
Ok got it !